src/mapping/FGreedyCutSolver.hpp

```diff
@@ -0,0 +1,243 @@
+
+  /// Maps the given input data
+  Eigen::VectorXd solveConservative(const Eigen::VectorXd &inputData, Polynomial polynomial) const;
+
+  // Clear all stored matrices
+  void clear();
+
+  // Returns the size of the input data
+  Eigen::Index getInputSize() const;
+
+  // Returns the size of the input data
+  Eigen::Index getOutputSize() const;
+
+private:
+  precice::logging::Logger _log{"mapping::FGreedySolver"};
+
+  std::pair<int, double> select(const Eigen::VectorXd &residual) const;
+
+  Eigen::MatrixXd buildEvaluationMatrix(const std::vector<int> &greedyIDs) const;
+  void updateKernelVector(const std::vector<int> greedyIDs, Eigen::VectorXd &kernelVector, const mesh::Vertex &x) const;
+  Eigen::VectorXd recalculateResidual(const std::vector<int> &greedyIDs, const Eigen::MatrixXd &cut, Eigen::MatrixXd &kernelMatrix, const Eigen::VectorXd &inputData, Eigen::VectorXd &r) const;
+
+  /// max iterations
+  const int _maxIter = 2000;
+
+  /// n_randon
+  const double _tolF = 1e-10;
+
+  Eigen::Index    _inSize  = 0;
+  Eigen::Index    _outSize = 0;
+  Eigen::MatrixXd _kernelEval;
+
+  const mesh::Mesh::VertexContainer _inputVertices;
+  const mesh::Mesh::VertexContainer _outputVertices;
+  RADIAL_BASIS_FUNCTION_T _basisFunction;
+  int _basisSize;
+  std::array<bool, 3> _activeAxis;
+};
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+std::pair<int, double> FGreedySolver<RADIAL_BASIS_FUNCTION_T>::select(const Eigen::VectorXd &residual) const
+{
+  Eigen::Index maxIndex;
+  double maxValue = residual.maxCoeff(&maxIndex);
+  return {maxIndex, maxValue};
+}
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::MatrixXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::buildEvaluationMatrix(const std::vector<int> &greedyIDs) const
+{
+  Eigen::MatrixXd matrixA(greedyIDs.size(), _outputVertices.size());
+
+  for (size_t i = 0; i < greedyIDs.size(); i++) 
+  {
+    const auto &u = _inputVertices.at(greedyIDs.at(i)).rawCoords();
+    for (size_t j = 0; j < _outputVertices.size(); j++) 
+    {
+      const auto  &v = _outputVertices.at(j).rawCoords();
+      const double squaredDifference = computeSquaredDifference(u, v, _activeAxis);
+      matrixA(i, j) = _basisFunction.evaluate(std::sqrt(squaredDifference));
+    }
+  }
+
+  return matrixA;
+}
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::recalculateResidual(const std::vector<int> &greedyIDs, const Eigen::MatrixXd &cut, Eigen::MatrixXd &kernelMatrix, const Eigen::VectorXd &inputData, Eigen::VectorXd &c) const {
+  int n = greedyIDs.size();
+  const auto &v = _inputVertices.at(greedyIDs.at(n-1)).rawCoords();
+  for (size_t i = 0; i < _inSize; i++) 
+  {
+    const auto &u = _inputVertices.at(i).rawCoords();
+    const double squaredDifference = computeSquaredDifference(u, v, _activeAxis);
+    kernelMatrix(i, n-1) = _basisFunction.evaluate(std::sqrt(squaredDifference));
+  }
+  const double a = (cut.block(n-1,0,1,n) * inputData(greedyIDs))(0,0);
```

Comment:
Is `(0,0)` just accessing the very first element or referring to the starting row and col?

Reply:
Converts the 1x1 Matrix into a double. Replaced with .value().